### DISCUSSION TITLE
Changing an form object property name instead of adding as new one

### DISCUSSION THREAD

**Discussion by cl-yufeng-rnd** [react-hook-form/react-hook-form#5310] [Q&A] [answered]
Hi all

I'm just started learning react and i'm trying to create a basic form using react-hook-form.
Just wanna ask if there is possible to change an form object **property name** instead of adding as a new one.

For example default values on initial render:

```
defaultValues: {
      person: {
        male: {
          firstName: "Jason",
          lastName: "Meh",
        }
      }
    }
```
Then from a dropdownlist with options 'male' and 'female', i choose "female" as the option, the property name changes:
```
defaultValues: {
      person: {
        female: {
          firstName: "Jason",
          lastName: "Meh",
        }
      }
    }
```

not adding new one like the one below:

```
defaultValues: {
      person: {
         male: {
          firstName: "Jason",
          lastName: "Meh",
        },
        female: {
          firstName: "Jason",
          lastName: "Meh",
        }
      }
    }
```

**Comment by thanh-nguyen-95** [comment, marked as answer]
I think this schema is a better approach:

```js
{
  person: {
    gender: 'male' | 'female',
    firstName: 'Jason',
    lastName: 'Meh'
  }
}
```

As for your question, it's not impossible but for something who just started learning _React_, it makes the form much more complicated than it should be in my opinion.

**Comment by ismyhairnice** [comment]
Hi, my school project actually also have the same requirement as OP.  I tried to use reset method to change the data structure but the form seems to not load with the new data structure with the property name changes:

Using example from OP:
reset({
   person: {
      female: {
         firstName: “Jason”,
         lastName: “Meh”
      }
})

Anyone have any idea if reset is the correct way to do this?
